ledger/tracker.go

```diff
@@ -456,6 +473,15 @@ func (tr *trackerRegistry) waitAccountsWriting() {
 	tr.accountsWriting.Wait()
 }
 
+func (tr *trackerRegistry) isBehindCommittingDeltas() bool {
+	if tr.accts.numDeltas() < tr.maxAccountDeltas {
+		return false
+	}
+
+	// there is a large number of deltas check if commitSyncer is not writing accounts
+	return tr.accountsCommitting.Load()
```

Comment:
Why do you want to return true only when the accounts are being committed? Even if they are not being actively committed, can't the deltas still grow too large?

Reply:
there are two considerations:
1. deltas grow only if they are not flushed, and they are not flushed 1) if the commit syncer is slow/blocked or 2) no new blocks added.
2. if relying only on deltas size it is possible to stall ledger: say we have 300 deltas after the last flush, and catchup cannot add more blocks because 300 is greater than the max. No new blocks added => these deltas are never flushed. Adding "actively committing" condition into the mix catchup service will be able to add more blocks when commitSyncer is done.

Reply:
Thanks for the explanation

Reply:
> Catchup will hit this and keep downloading blocks as fast as it can, beyond the 256 limit

As soon as one more block added into ledger the condition becomes true and catchup exits.

> Or you just have to wait for catchup to stop then start up again?

Correct, on the new iteration it will add new blocks.

> Assume this accountsCommitting didn't exist — you hit 256 deltas, catchup service stops. commitSyncer flushes.

It might not flush because flushes are triggered by new blocks.